timesketch/lib/llms/features/llm_forensic_report.py

```diff
@@ -0,0 +1,304 @@
+        namespace=METRICS_NAMESPACE,
+    ),
+    "llm_forensic_report_unique_events_total": prometheus_client.Counter(
+        "llm_forensic_report_unique_events_total",
+        "Total number of unique events sent to the LLM for forensic report generation",
+        ["sketch_id"],
+        namespace=METRICS_NAMESPACE,
+    ),
+    "llm_forensic_report_stories_created_total": prometheus_client.Counter(
+        "llm_forensic_report_stories_created_total",
+        "Total number of forensic report stories created",
+        ["sketch_id"],
+        namespace=METRICS_NAMESPACE,
+    ),
+}
+
+
+class LLMForensicReportFeature(LLMFeatureInterface):
+    """LLM Forensic Report feature."""
+
+    NAME = "llm_forensic_report"
+    PROMPT_CONFIG_KEY = "PROMPT_LLM_FORENSIC_REPORT"
+    RESPONSE_SCHEMA = {
+        "type": "object",
+        "properties": {
+            "summary": {
+                "type": "string",
+                "description": "Detailed forensic report summary of the events",
+            }
+        },
+        "required": ["summary"],
+    }
+
+    def _get_prompt_text(self, events_dict: list[dict[str, Any]]) -> str:
+        """Reads the prompt template from file and injects events.
+        Args:
+            events_dict: List of event dictionaries to inject into prompt.
+        Returns:
+            Complete prompt text with injected events.
+        Raises:
+            ValueError: If the prompt path is not configured or placeholder is missing.
+            FileNotFoundError: If the prompt file cannot be found.
+            IOError: If there's an error reading the prompt file.
+            OSError: If there's an error reading the prompt file.
+        """
+        prompt_file_path = current_app.config.get(self.PROMPT_CONFIG_KEY)
+        if not prompt_file_path:
+            logger.error("%s config not set", self.PROMPT_CONFIG_KEY)
+            raise ValueError("LLM forensic report prompt path not configured.")
+
+        try:
+            with open(prompt_file_path, "r", encoding="utf-8") as file_handle:
+                prompt_template = file_handle.read()
+        except FileNotFoundError as exc:
+            logger.error("Forensic report prompt file not found: %s", prompt_file_path)
+            raise FileNotFoundError(
+                f"LLM Prompt file not found: {prompt_file_path}"
+            ) from exc
+        except IOError as e:
+            logger.error("Error reading prompt file: %s", e)
+            raise IOError("Error reading LLM prompt file.") from e
+
+        if "<EVENTS_JSON>" not in prompt_template:
+            logger.error("Prompt template is missing the <EVENTS_JSON> placeholder")
+            raise ValueError(
+                "LLM forensic report prompt template is missing the "
+                "required <EVENTS_JSON> placeholder."
+            )
+
+        prompt_text = prompt_template.replace("<EVENTS_JSON>", json.dumps(events_dict))
+        return prompt_text
+
+    def _run_timesketch_query(
```

Comment:
As discussed 1:1 have a look if you can replace this with the [OpenSearchDataStore.search()](https://github.com/google/timesketch/blob/master/timesketch/lib/datastores/opensearch.py#L581) function instead.